reframe/utility/typecheck.py

```diff
@@ -3,62 +3,89 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
-'''Dynamic recursive type checking of aggregate data structures.
+'''Dynamic recursive type checking of collections.
 
-This module defines types for aggregate data structures, such as lists,
-dictionaries etc. that you can use with the ``isinstance`` builtin function to
-recursively check of all the elements of an aggregate data structure.
-Suppose you have a list of integers, suchs as ``[1, 2, 3]``, the following
-checks should hold:
+This module defines types for collections, such as lists, dictionaries etc.
```

Comment:
```suggestion
This module defines types for collections, such as lists, dictionaries etc,
```

Reply:
The comma should just be added here, not replace the `.`.